src/web3-adapter/eth_gasStation.ts

```diff
@@ -0,0 +1,29 @@
+import https from "https";
+
+export function patchEthGasStation(web3: any): void {
+  web3.ethGasSation = (apiKey: string) => {
```

Comment:
I think we can be more clear about how we expose this. Eth Gas Station has multiple endpoints (well, two), but from the method name here it's not obvious which one is being called. Also, this isn't named consistently with other web3 methods, which are usually verbs, e.g. `getBlockNumber`. We might want to put it at `web3.ethGasStation.getGasPrice` instead.

I also find it kinda odd to be passing in the API key on each request, but given this setup it can't really be avoided. Perhaps this is a sign that we're trying to tack on features to alchemy-web3 that don't really fit- a more natural API would be a separate Eth Gas Station client, which is initialized with the API key and then calls the methods as needed. This would have the advantage that you'd be able to request info from Eth Gas Station without having an Eth provider url (technically you could initialize the alchemy-web3 object with a dummy url, but that's not good UX).

I'm not opposed to awkwardly tacking on features if it's something that we know will provide value to users. Have we been hearing that this is a desired feature?